src/distances.jl

```diff
@@ -1,27 +1,27 @@
 # ------------------------------------------------------------------
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
-
-# flip arguments so that points always come first
-evaluate(d::PreMetric, g::Geometry, p::Point) = evaluate(d, p, g)
-
 """
-    evaluate(Euclidean(), point, line)
+    mindistance(metric::PreMetric, g::Union{Geometry,Point}, p::Point)
```

Comment:
The function signature should be more intuitive, meaning we shouldn't pass a metric as the first argument. We should instead have something like

```julia
mindistance(p::Point, g::Geometry; metric=Euclidean())
```
that focuses on the objects and leaves the metric as an option.

Also notice that we should implement a version with `p` as the first argument always and flip the argument to handle the symmetry. This is for organizational purposes and consistency across the codebase.

Reply:
> Also notice that we should implement a version with p as the first argument always and flip the argument to handle the symmetry. This is for organizational purposes and consistency across the codebase.

This is implemented in line 13:
`mindistance(metric::PreMetric, p::Point, g::Geometry) = mindistance(metric, g, p)`

Reply:
> Also, we may need to think more carefully about the function name when one of the arguments is a point because there is no much use in other type of distance other than the `mindistance`. So we could probably call it `distance` directly. This `mindistance` versus `maxdistance` issue only becomes relevant when we have two geometries as the arguments I guess.

I would argue that `mindistance` vs. `maxdistance` is also relevant as soon as one of the objects is not a singleton set (i.e. a Point). However, I am somewhat convinced that `maxdistance` almost never makes sense. For example, for `Line`, `Ray`, and `Plane` `maxdistance` would have to return `Inf`.

I would therefore vote for only implementing `mindistance` (and maybe call it just `distance`) and worry about other types of distances if someone actually needs them.